File: Diamond.cpp
#include<iostream>
using namespace std;

class ClassA
{
    public:
        void aClass()
        {
            cout<<"I am a ClassA\n";
        }
};

class ClassB: public ClassA
{
    public:
        void bClass()
        {
            cout<<"I am a ClassB\n";
        }
};

class ClassC: public ClassA
{
    public:
        void cClass()
        {
            cout<<"I am a ClassC\n";
        }
};

class ClassD: public ClassC,public ClassB
{
    public:
        void dClass()
        {
            cout<<"I am a ClassD\n";
        }
};

int main()
{
    ClassD obj1;
    obj1.bClass();
    obj1.cClass();
    obj1.dClass();
    return 0;
}

File: Diamond.java
class ClassA
{
    public
        void aClass()
        {
            System.out.print("Iam a ClassA"+"\n");
        }
}

class ClassB extends ClassA
{
    public
        void bClass()
        {
            System.out.print("Iam a ClassB"+"\n");
        }
}

class ClassC extends ClassA
{
    public
        void cClass()
        {
            System.out.print("Iam a ClassC"+"\n");
        }
}



public class DiamondProblem 
{
    public static void main(String[] args)
    {
        System.out.print("Diamond Problem"+"\n");
        ClassB class1=new ClassB();
        ClassC class2=new ClassC();
        class1.aClass();
        class1.bClass();
        class2.aClass();
        class2.cClass();
        
    }
}
